IPG-Automotive-UK/react-ui
# TD-3228 Create `PrototypePreview` component

Closes [TD-3228](https://sce.myjetbrains.com/youtrack/issue/TD-3228)
Contributes [TD-3196](https://sce.myjetbrains.com/youtrack/issue/TD-3196)

## Changes




A brief description of what this pull request solves / introduces.

- Creates a new component `PrototypePreview`
- Updates the `StatusIcon` component to expose the padding prop, so this can be adjusted
- Updated the `StatusLabel` component to expose the `gap` prop, so this can be adjusted.
- Updates the tests for `StatusIcon` and `StatusLabel` to test the added functionality
- Adds tests for `PrototypePreview`

Maybe a short description of what is not included and will come in future work where appropriate.

## Dependencies




## UI/UX

[FIGMA](https://www.figma.com/design/WoHtIQKM6eLlOUuzK1zlpB/VIRTO.TEST?node-id=1789-10490&node-type=canvas&t=KdU30i7WnfrwS13F-0)

Default light:
![image](https://github.com/user-attachments/assets/1376b9e8-ab61-44c7-901f-e784854cbba9)

Default dark:
![image](https://github.com/user-attachments/assets/8f51cc4f-e9ba-4b88-b18e-d94d23b8527c)

Overflow:
![image](https://github.com/user-attachments/assets/88192e6c-51ce-4f1f-b87b-701b4a79ab9f)





## Testing notes
1. Load up the storybook from the CI/CD GitHub process. Navigate to `PrototypePreview`
2. Make sure the component matches the [FIGMA](https://www.figma.com/design/WoHtIQKM6eLlOUuzK1zlpB/VIRTO.TEST?node-id=1789-10490&node-type=canvas&t=KdU30i7WnfrwS13F-0)
3. Play around with the entries to the component. Make sure that all text overflow is controlled. Long text should truncate and nothing should clash.
4. Make sure clicking on the Prototype name opens a link in a new tab.



## Author checklist

Before I request a review:



~- [ ] I have populated the deploy-preview with relevant test data.~
